src/multicellular_stage/editor/CellBodyPlanEditorComponent.cs

```diff
@@ -477,13 +514,47 @@ public bool ShowCellOptions()
         return true;
     }
 
-    protected CellType CellTypeFromName(string name)
+    public Dictionary<Compound, float> GetAdditionalCapacities(out float nominalCapacity)
     {
-        return Editor.EditedSpecies.CellTypes.First(c => c.TypeName == name);
+        // TODO: merge this with nominal get to make this more efficient
+        return CellBodyPlanInternalCalculations.GetTotalSpecificCapacity(editedMicrobeCells.Select(o => o.Data!),
+            out nominalCapacity);
     }
 
-    protected override void OnTranslationsChanged()
+    public void OnCurrentPatchUpdated(Patch patch)
     {
+        CalculateEnergyAndCompoundBalance(editedMicrobeCells);
+
+        organismStatisticsPanel.UpdateLightSelectionPanelVisibility(
+            Editor.CurrentGame.GameWorld.WorldSettings.DayNightCycleEnabled && Editor.CurrentPatch.HasDayAndNight);
+    }
+
+    public override void OnLightLevelChanged(float dayLightFraction)
+    {
+        var maxLightLevel = Editor.CurrentPatch.Biome.GetCompound(Compound.Sunlight, CompoundAmountType.Biome).Ambient;
+        var templateMaxLightLevel =
+            Editor.CurrentPatch.GetCompoundAmountForDisplay(Compound.Sunlight, CompoundAmountType.Template);
+
+        // Currently, patches whose templates have zero sunlight can be given non-zero sunlight as an instance. But
```

Comment:
Seems like a pretty long explanation to copy-paste so I think a some kind of refactoring is needed to have this long comment and bug "workaround" in a single place in the code base.